src/OrchardCore/OrchardCore.Infrastructure/Filters/ApiExceptionHandlingFilter.cs

```diff
@@ -0,0 +1,43 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.Mvc.Filters;
+using System.Text.Json;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Http.Json;
+using Json.More;
+namespace OrchardCore.Infrastructure.Filters;
+
+public class ApiExceptionHandlingFilter : IExceptionFilter
+{
+    private static readonly JsonSerializerOptions _jsonOptions = new()
+    {
+        PropertyNamingPolicy = JsonNamingPolicy.CamelCase,
+        WriteIndented = true
+    };
+
+    public ApiExceptionHandlingFilter()
+    {
+
+    }
+
+    public void OnException(ExceptionContext context)
+    {
+        var response = new ProblemDetails
+        {
+            Type = "https://tools.ietf.org/html/rfc7231#section-6.6.1",
+            Title = "An error occurred while processing your request.",
+            Status = StatusCodes.Status500InternalServerError,
```

Comment:
I think I would use a 400 for this one.

Reply:
We could also pass an error code along with the error string that we could document. Just to make a distinction between 500 and 400.